exercises/dotnet/encryption-context-start/src/App.cs

```diff
@@ -0,0 +1,155 @@
+            {
+                Generator = Config.FaytheKmsKeyId,
+                KmsKeyIds = new List<string>() {Config.WalterKmsKeyId}
+            });
+
+            Api api = new(amazonDynamoDBClient, tableName, amazonS3Client, bucketName, keyring);
+
+            Console.CancelKeyPress += new ConsoleCancelEventHandler(CancelHandler);
+            while (_continue)
+            {
+                Console.WriteLine("Enter a number to chose a menu option:");
+                Console.WriteLine("1. List");
+                Console.WriteLine("2. Store");
+                Console.WriteLine("3. Retrieve");
+                try
+                {
+                    switch (Console.ReadLine())
+                    {
+                        case "1":
+                            var items = await api.List();
+                            await ListToConsole(api, items);
+                            break;
+                        case "2":
+                            Console.WriteLine("Enter data to store:");
+                            var data = Console.ReadLine();
+                            if (data is null || data == "")
+                            {
+                                Console.WriteLine("You must enter a valid input!");
+                                break;
+                            }
+                            Console.WriteLine($"Storing: {data}");
+                            var context = ReadEncryptionContext();
+                            if (context.Count > 0)
+                            {
+                                Console.WriteLine("Adding Encryption Context:");
+                                Console.WriteLine(string.Join(Environment.NewLine, context));
+                            }
+                            await api.Store(Encoding.UTF8.GetBytes(data), context);
+                            Console.WriteLine("Data stored successfully!");
+                            break;
+                        case "3":
+                            Console.WriteLine("Enter key of item to retrieve:");
+                            var key = Console.ReadLine();
+                            if (key is null || key == "")
+                            {
+                                Console.WriteLine("You must enter a valid input!");
+                                break;
+                            }
+                            Console.WriteLine("Enter the Encryption Context for this object. The Encryption Context must match what was provided when the object was stored.");
+                            var retrievalContext = ReadEncryptionContext();
+                            Console.WriteLine($"Retrieving with Encryption Context: {key}");
+                            Console.WriteLine(string.Join(Environment.NewLine, retrievalContext));
+                            var documentBundle = await api.Retrieve(key);
```

Comment:
This is the only blocking change needed
```suggestion
                            var documentBundle = await api.Retrieve(key, retrievalContext);
```